lms/djangoapps/student_account/views.py

```diff
@@ -74,6 +74,23 @@ def login_and_registration_form(request, initial_mode="login"):
     if request.user.is_authenticated():
         return redirect(redirect_to)
 
+    running_pipeline = pipeline.get(request)
+    if running_pipeline and initial_mode == 'register':
+        current_provider = third_party_auth.provider.Registry.get_from_pipeline(running_pipeline)
+        enterprise_customer = enterprise_customer_for_request(request)
+        if enterprise_customer and current_provider and current_provider.sync_learner_profile_data:
+            user_details = running_pipeline['kwargs']['details']
+
+            try:
+                # pylint: disable=invalid-name
+                qs = {'email': user_details['email']} if user_details.get('email') else \
+                    {'username': user_details.get('username')}
+                user = User.objects.get(**qs)
+            except User.DoesNotExist:
+                pass
+            else:
+                return redirect(reverse('signin_user'), next=redirect_to)
```

Comment:
Where does the user data sync happen in this scenario where the user account already exists?

Reply:
It seems like this logic should be somewhere in the TPA pipeline and not here.

Reply:
Yes, Uman added [this functionality in the pipeline](https://github.com/edx/edx-platform/blob/master/common/djangoapps/third_party_auth/pipeline.py#L719) as part of https://github.com/edx/edx-platform/pull/16624.

Reply:
it will be synced in the pipeline in case of user login.

Reply:
This block seems like it should be a TPA pipeline partial. Is there a reason it can't be?

Reply:
See the [edx-platform custom TPA pipeline code](https://github.com/edx/edx-platform/blob/master/common/djangoapps/third_party_auth/pipeline.py) and the [python-social-auth pipeline code](https://github.com/python-social-auth/social-core/blob/master/social_core/pipeline/social_auth.py#L51) for examples of what I am talking about. [Here](https://github.com/edx/edx-platform/blob/master/common/djangoapps/third_party_auth/settings.py#L46) is the list of pipeline functions that we currently run in edx-platform.

Reply:
@douglashall I tried the pipeline before implementing this but with pipeline we cannot enforce the user to not create an account when one already exists. So, we need this login in the `login_and_registration_form` view so that we can enforce that if learner account already exists we redirect learner back to login page.

Reply:
Sounds like you weren't injecting this logic early enough in the pipeline. Do you have a branch with the pipeline approach? I think it would be better to explore that a bit further.

Reply:
I do not have a branch with the changes but I will recreate the changes in a branch and see if we can make it work.

Reply:
I have made the changes in a [separate branch](https://github.com/edx/edx-platform/commit/a640295d9a0e3e16178a45c41f82fb976bff4ed7) but this way user is redirected to login page and the pipeline breaks and learner is not associated with the identity provider or the enterprise customer.